mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -475,13 +476,13 @@ def _get_eager_run_query_options():
                  run attributes when fetching a run: ``latest_metrics``, ``params``, and ``tags``.
         """
         return [
-            # Use a subquery load rather than a joined load in order to minimize the memory overhead
-            # of the eager loading procedure. For more information about relationship loading
-            # techniques, see https://docs.sqlalchemy.org/en/13/orm/
+            # Use a select in load rather than a joined load in order to minimize the memory
+            # overhead of the eager loading procedure. For more information about relationship
+            # loading techniques, see https://docs.sqlalchemy.org/en/13/orm/
             # loading_relationships.html#relationship-loading-techniques
-            sqlalchemy.orm.subqueryload(SqlRun.latest_metrics),
-            sqlalchemy.orm.subqueryload(SqlRun.params),
-            sqlalchemy.orm.subqueryload(SqlRun.tags),
+            sqlalchemy.orm.selectinload(SqlRun.latest_metrics),
+            sqlalchemy.orm.selectinload(SqlRun.params),
+            sqlalchemy.orm.selectinload(SqlRun.tags),
```

Comment:
Can you provide some context for this change? Is this compatible with SQL Server? 

```
The only scenario in which selectin eager loading is not feasible is when the model is using composite primary keys, and the backend database does not support tuples with IN, which currently includes SQL Server.
```

(from https://docs.sqlalchemy.org/en/13/orm/loading_relationships.html#select-in-loading)

Reply:
@dianacarvalho1 Got it! Thank you for confirming! The fact that existing CI passes for SQL Server is encouraging. To provide an added level of confidence, can we introduce an `mlflow.search_runs()` operation to the test coverage here: https://github.com/mlflow/mlflow/commit/d44567feb2d17f328d6d0a580948882c3d9b9d5c#diff-a506d62cddeec457c1fa4cb3e713126a72d968fb20c94a92454e3925d94e6fd2R35?

Reply:
@dbczumar I'm not sure what you mean. Something like:
````
def run_search_runs():
    runs = mlflow.search_runs(
      experiment_ids="0",
      order_by=["param.start_time DESC"]
    )

    run = mlflow.get_run(runs["run_id"][0)

````

Reply:
@dianacarvalho1 Yes, can we add the following right below https://github.com/mlflow/mlflow/blob/20a7d6baaa9b040bb5e39b770aaae81ae0af0ca2/tests/db/run_checks.py#L35 in the same method?

```
    runs = mlflow.search_runs(
      experiment_ids="0",
      order_by=["param.start_time DESC"]
    )

    run = mlflow.get_run(runs["run_id"][0)
```